handlers/handler-publickey-auth/src/main/java/io/servicecomb/authentication/consumer/RSACoumserTokenManager.java

```diff
@@ -0,0 +1,81 @@
+package io.servicecomb.authentication.consumer;
+
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.SignatureException;
+import java.security.spec.InvalidKeySpecException;
+import java.util.Date;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.servicecomb.authentication.RSAAuthenticationToken;
+import io.servicecomb.foundation.common.utils.RSAUtils;
+import io.servicecomb.foundation.token.RSAKeypair4Auth;
+import io.servicecomb.serviceregistry.RegistryUtils;
+
+public class RSACoumserTokenManager {
+
+  private static final Logger logger = LoggerFactory.getLogger(RSACoumserTokenManager.class);
+
+  private ReadWriteLock readWriteLock = new ReentrantReadWriteLock();
+
+  private RSAAuthenticationToken token;
+
+  public String getToken() {
+    readWriteLock.readLock().lock();
+    if (isvalid(token)) {
+      String tokenStr = token.format();
+      readWriteLock.readLock().unlock();
+      return tokenStr;
+    } else {
+      readWriteLock.readLock().unlock();
+      return createToken();
```

Comment:
it seems you are writing a double check logic.
if true, then your code is too complex.